File: project/src/shared.ts
import fetch from 'node-fetch'

export interface Params {
  symbol: string;
}

export interface ForecastOutput {
  symbol: string,
  object_key: string,
}

export interface ProcessResultInput {
  symbols: string[],
  object_keys: string[],
}
export interface ProcessResultOutput {
  labels: string[],
  datasets: { label: string, data: number[] }[]
}

export type CreateChartInput = ProcessResultOutput
export interface CreateChartOutput {
  url: string,
}

export function assert(value: string): asserts value {
  if (!value) {
    throw new Error('A symbol must be specified')
  }
}

export async function getIbmBucketObject(objectKey: string): Promise<any> {
  const endpoint = process.env.IBM_OBJECT_STORAGE_ENDPOINT_URL
  const bucket = process.env.IBM_OBJECT_STORAGE_BUCKET_NAME

  const url = `https://${endpoint}/${bucket}/${objectKey}`

  let object = await fetch(url)
  return await object.json()
}

File: project/src/shared.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize, Serialize)]
pub struct FetchPricesInput {
  pub symbol: String,
}
#[derive(Debug, Deserialize, Serialize)]
pub struct FetchPricesOutput {
  pub symbol: String,
  pub object_key: String,
}

pub type ForecastInput = FetchPricesOutput;
pub type ForecastOutput = ForecastInput;

pub type ProcessResultInput = ForecastOutput;
#[derive(Debug, Deserialize, Serialize)]
pub struct Dataset {
  pub label: String,
  pub data: Vec<f32>,
}
#[derive(Debug, Deserialize, Serialize)]
pub struct ProcessResultOutput {
  pub labels: Vec<String>,
  pub datasets: Vec<Dataset>,
}

pub type CreateChartInput = ProcessResultOutput;
#[derive(Debug, Deserialize, Serialize)]
pub struct CreateChartOutput {
  pub url: String,
}
